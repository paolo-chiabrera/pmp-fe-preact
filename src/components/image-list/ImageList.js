import { h } from 'preact';
import { Link } from 'preact-router/match';

export default function ImageList({ images }) {
	const imagesList = images.map(image => {
		const { filename } = image;

		return (
			<Link href={ `/images/${filename}` }>
				<li>{ filename }</li>
			</Link>
		);
	});

	return (
		<ul class="image-list">
			{ imagesList }
		</ul>
	);
}