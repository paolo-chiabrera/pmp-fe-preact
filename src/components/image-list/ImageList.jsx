import { h } from 'preact';
import { Link } from 'preact-router/match';

import './style.scss';

const IMAGE_SIZE = 150;

export default function ImageList({ images, pageNumber }) {
  const imagesList = images.map(image => {
    const { filename, meta: { ext, label, ratio } } = image;

    const imageWrapperStyle = {
      backgroundImage: `url(http://images.picmeplease.eu/images/${filename}.${ext})`
    };

    const flexGridItemStyle = {
      height: `${IMAGE_SIZE}px`,
      width: `${ratio * IMAGE_SIZE}px`
    };

    return (
      <div class="flex-grid-item" key={filename} style={flexGridItemStyle}>
        <Link href={`/images/${filename}`}>
          <div class="image-wrapper" style={imageWrapperStyle} key={filename}/>
        </Link>
      </div>
    );
  });

  return (
    <div class="flex-grid" key={`flex-grid-${pageNumber}`}>
      {imagesList}
    </div>
  );
}
