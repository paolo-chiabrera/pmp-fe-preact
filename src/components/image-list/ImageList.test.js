import { h } from 'preact';

import render from 'preact-render-to-string';

import { StaticRouter } from 'react-router';

import ImageList from './ImageList.jsx';

describe('Given a ImageList component', () => {
  it('should render correctly', () => {
    const tree = render(
      <StaticRouter location="/">
        <ImageList images={[]} pageNumber={0} />
      </StaticRouter>
    );

    expect(tree).toMatchSnapshot();
  });
});
