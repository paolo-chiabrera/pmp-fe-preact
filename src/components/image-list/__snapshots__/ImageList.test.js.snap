// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a ImageList component should render correctly 1`] = `"[object Object]"`;
